### PR TITLE
Rework filesystem mock

### PR DESCRIPTION
**Pull request by gchazot**
Suggestion of a change in tests description of local filesystem contents to normalise the structure. This allows to remove redundancy of data, and will allow simplification of the code handling filesystem mocking.

### CODE REVIEW

File: tests/util.py [hunk 1]
```diff
@@ -229,11 +229,19 @@ def __exit__(self, *args, **kwargs):
...
-            patch_directories,
-        )
+    def __init__(self, patch_files, patch_directories, local_files):
+        if local_files is None:
+            directories, files, files_data = self._parse_files_and_directories(
+                patch_files,
+                patch_directories,
+            )
+        else:
+            print(self._parse_files_and_directories(
```
Comment (Fizzadar) [inline tests/util.py]: Are these leftovers or to be used when debugging tests (if so can we add a quick comment indicating)?

**Review by Fizzadar — COMMENTED** [review]
BIG fan of this - a much cleaner way to specify the dirs/files and much less error prone 👍

File: tests/util.py (hunk 1 above)
Comment (gchazot) [reply tests/util.py]: Oops, these are leftovers indeed 59fb3e3

**Comment by gchazot** [comment]
Cool, when I get the time I'll finalise the migration of all tests and remove the old format

**Comment by gchazot** [comment]
I've cleaned this up, it's ready now

**Review by Fizzadar — APPROVED** [review]
Thank you for putting this together @gchazot!

**Pull request merged**